porcupine/plugins/highlight/tree-sitter-data/token-mappings/python.yml

```diff
@@ -0,0 +1,204 @@
+dont_recurse_inside:
+  - string  # TODO: recurse inside f-strings
+  # TODO: distinguish between variables and attributes?
+  # Can't just add "attribute" to dont_recurse_inside because then the object
+  # whose attribute it is won't highlight, e.g. open("foo", "rb").read()
+
```

Comment:
Yess, this is what I wanted to suggest. With all this stuff I found the filetype config file quite messy.


My other issue is this:
`pygments`:
![image](https://user-images.githubusercontent.com/77941087/183312258-9e5c0dde-b181-4e37-aaff-a6aea5c35196.png)
`tree-sitter`:
![image](https://user-images.githubusercontent.com/77941087/183312266-78a5ac1b-c094-4c4f-b288-85a866c10f7d.png)

Syntax highlighting looks quite obscure without punctuation and operator highlighting, so I'd like to fix that.
@Akuli do you think this PR is appropriate for that, or should I open a new one, after this is completed?

Reply:
I'm fine with it either way. I recently moved the config to a yaml file. If you can figure out a way to get this to work, just push your commits to this pr :)

Reply:
lol, this PR has gotten crazy huge: it adds more than 1k lines already. I think I'll get this to pass CI and merged pretty soon, and add all the remaining things in separate PRs.